src/Resource/Ssl/Ssl.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * @package Nexcess-SDK
+ * @license https://opensource.org/licenses/MIT
+ * @copyright 2018 Nexcess.net, LLC
+ */
+
+declare(strict_types  = 1);
+
+namespace Nexcess\Sdk\Resource\Ssl;
+
+use Nexcess\Sdk\ {
+  Resource\Client\Client,
+  Resource\Model
+};
+
+/**
+ * Certificates
+ */
+class Ssl extends Model {
+
+  /** {@inheritDoc} */
+  public const MODULE_NAME = 'Ssl';
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_ALIASES = [
+    'id' => 'cert_id'
+  ];
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_COLLAPSED = [];
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_MODELS = [
+    'client' => Client::class
+  ];
+
+  /** {@inheritDoc} */
+  protected const _PROPERTY_NAMES = [
+    'alt_names',
+    'approver_email',
+    'cert_id',
+    'chain',
+    'chain_crts',
+    'client_id',
+    'common_name',
+    'crt',
+    'csr',
+    'distinguished_name',
+    'domain',
+    'id',
+    'key',
+    'months',
+    'package_id'
+  ];
+
+  /** {@inheritDoc} */
+  protected const _READONLY_NAMES = [
+    'client_id',
+    'identity',
+    'is_expired',
+    'is_installable',
+    'is_multi_domain',
+    'is_real',
+    'is_wildcard',
+    'valid_from_date',
+    'valid_to_date'
+  ];
+
+  /**
+   * Create a new backup
+   * This method has two modes. If a CSR is present then it will attempt to
+   * create the cert via the CSR. Otherwise, it will attempt to create the csr
+   * and key and then create the certificate.
+   *
+   * @return Ssl
+   * @throws \GuzzleHttp\Exception\ClientException on fail
+   */
+  public function create() : Ssl {
+    $endpoint = $this->_getEndpoint();
+    assert($endpoint instanceof Endpoint);
+
+    if (!empty($this->get('csr'))) {
```

Comment:
`csr` is checked, but no other values are. If they're missing, we'll get a `TypeError`, which is unexpected when invoking a method with no arguments. Should we check first and throw?

Reply:
not in favor of it, no. We already validate the parameters at the endpoint. That having been said, if you feel strongly about it, we can.

Reply:
the check for csr is to determine which to create. I don't really care about the CSR itself, it's just a signal at this point. It could contain HELLO WORLD and we'd try to use it.